src/main/java/homework10/MyHashMap.java

```diff
@@ -0,0 +1,148 @@
+package homework10;
+
+import java.util.Map;
+
+public class MyHashMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    Node<K, V>[] arrayOfNodes;
+    int size;
+
+    public MyHashMap() {
+    }
+
+    public V get(K key) {
+        return getNode(key).getValue();
+    }
+
+    public V put(K key, V value) {
+        if ((arrayOfNodes == null || arrayOfNodes.length == 0) ^ size == arrayOfNodes.length * LOAD_FACTOR) {
+            arrayOfNodes = resize(arrayOfNodes);
+        }
+        Node<K, V> positionNode = arrayOfNodes[hash(key) % arrayOfNodes.length];
+        if (positionNode == null) {
+            arrayOfNodes[hash(key) % arrayOfNodes.length] = new Node<>(key, value, null);
+            size++;
+        } else if (positionNode.getKey().equals(key) && positionNode.hash == hash(key)) {
+            V oldValue = positionNode.getValue();
+            arrayOfNodes[hash(key) % arrayOfNodes.length].setValue(value);
+            return oldValue;
+        } else if (!positionNode.getKey().equals(key) && positionNode.hash == hash(key)) {
+            Node<K, V> nextInBucket = positionNode.next;
+            if (nextInBucket != null) {
+                do {
+                    if (nextInBucket.key.equals(key)) {
+                        V oldValue = nextInBucket.getValue();
+                        nextInBucket.setValue(value);
+                        return oldValue;
+                    }
+                } while (nextInBucket.next != null);
+            } else {
+                nextInBucket = new Node<>(key, value, null);
+                size++;
+            }
+        }
+        return null;
+    }
+
+    public Node<K, V> getNode(K key) {
+        Node<K, V> firstInBucket = arrayOfNodes[hash(key) % arrayOfNodes.length];
+        Node<K, V> nextInBucket = firstInBucket.next;
+        if (arrayOfNodes != null && arrayOfNodes.length > 0 && firstInBucket != null) {
+            if (firstInBucket.key == null && key == null || firstInBucket.key.equals(key)) {
+                return firstInBucket;
+            }
+            if (nextInBucket != null) {
+                do {
+                    if (nextInBucket.key.equals(key)) {
+                        return nextInBucket;
+                    }
+                } while (nextInBucket.next != null);
+            }
+        }
+        return null;
+    }
+
+    public Node<K, V>[] resize(Node<K, V>[] oldArrayOfNodes) {
```

Comment:
do we need use parameter here?
```suggestion
    private Node<K, V>[] resize(Node<K, V>[] oldArrayOfNodes) {
```